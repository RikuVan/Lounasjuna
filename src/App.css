.App {
  margin: 0 auto;
  max-width: 960px;
}

.App-header {
  padding: 3rem 0;
  text-align: center;
}

.App-subheader {
  background-color: rgba(119, 85, 102, 0.9);
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  display: flex;
  justify-content: flex-end;
  padding: 15px;
}

.App-subheader-links {
  flex: 1 1 auto;
}

.App-content {
  background-color: #fff;
}

h1 {
  font-family: 'Oleo Script';
  margin: 0;
}

.Logo {
  display: inline-block;
  line-height: 1;
  position: relative;
}

.Logo-train {
  color: transparent;
  font-size: 8rem;
  text-shadow: 0 0 0 rgba(255, 255, 255, 0.4);
  transform: scale(-1, 1);
}

.Logo-text {
  color: #fff;
  font-size: 4rem;
  left: 50%;
  position: absolute;
  text-shadow: 0 0 1rem rgba(0, 0, 0, 0.2);
  top: 0;
  transform: translate(-50%, 50%);
}
