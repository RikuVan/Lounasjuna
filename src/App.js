import React, {Component, PropTypes} from 'react'
import {connect} from 'react-redux'
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom'
import {bindActionCreators} from 'redux'
import {
  attemptSignInWithGoogle,
  cancelGoogleAuth
} from './actions/auth'
import {fetchUsers} from './actions/users'
import './App.css'
import RestaurantList from './views/restaurants/Restaurant-list'
import RestaurantForm from './views/forms/Restaurant-form'
import SignInOrOut from './components/Sign-in-out'
import CurrentUser from './components/Current-user'
import ButtonLink from './components/ButtonLink'
import {Link} from 'react-router-dom'

class App extends Component {
  componentDidMount() {
    this.props.fetchUsers()
  }

  handleSignIn = () => this.props.attemptSignInWithGoogle()

  handleSignOut = () => this.props.cancelGoogleAuth()

  render() {
    const {auth} = this.props
    const showSignIn = this.props.auth.status === 'ANONYMOUS'
    const awaitingLogin = auth.status === 'AWAITING_AUTH_RESPONSE'
    return (
      <Router>
        <div className="App">
          <div className="App-header">
            <Link to={'/'}>
              <h1 className="Logo">
                <div className="Logo-train">ðŸš‚</div>
                <div className="Logo-text">Lounasjuna</div>
              </h1>
            </Link>
          </div>
          {/******** START NAV *******/}
          <div className="App-subheader">
            {auth.status === 'SIGNED_IN' &&
              <div className="App-subheader-links">
                <ButtonLink path="/uusi" type="white">
                  <i className="fa fa-plus-circle" /> LisÃ¤Ã¤ lounaspaikka
                </ButtonLink>
              </div>
            }
            <CurrentUser {...auth} />
            <SignInOrOut
              type={showSignIn ? 'SignIn' : 'SignOut'}
              onClickHandler={showSignIn ? this.handleSignIn : this.handleSignOut}
              loading={awaitingLogin}
            />
            <CurrentUser {...auth} />
            <Route path="/" exact render={() =>
              /**
               * react router provides a render method instead of just passing
               * as component, inside of which you can add logic and/or
               * pass props
               */
              auth.status === 'SIGNED_IN' ?
                <ButtonLink className="Nav--button" path="/lounaspaikka/uusi">
                  {'LisÃ¤Ã¤ lounaspaikka'}
                </ButtonLink> : null
                }
            />
            <Route path="/lounaspaikka/" render={() =>
                <ButtonLink className="Nav--button" path="/">
                  {'PÃ¤Ã¤sivu'}
                </ButtonLink>
              }
            />
          </div>
          {/******** END NAV *******/}
          {/****** START CONTENT ******/}
          <div className="App-content">
            <Switch>
              <Route
                path="/"
                exact
                render={() =>
                  <RestaurantList auth={auth} />
                }
              />
              <Route
                path="/lounaspaikka/uusi"
                render={({history}) => {
                  /***
                   *  here react router gives us the history object
                   *  which allows us to perform programmatic routing
                   *  in the form component using 'push'
                   */

                  if (auth.status === 'SIGNED_IN') {
                    return <RestaurantForm history={history}/>
                  } else {
                    return <RestaurantList auth={auth}/>
                  }
                }}
              />
              <Route path="/" render={() => <h3>404</h3>} />
            </Switch>
          </div>
          {/******* END CONTENT *******/}
        </div>
      </Router>
    )
  }
}

App.propTypes = {
  auth: PropTypes.object.isRequired,
  attemptSignInWithGoogle: PropTypes.func.isRequired,
  cancelGoogleAuth: PropTypes.func.isRequired,
  fetchUsers: PropTypes.func.isRequired
}

const mapStateToProps = state => ({auth: state.auth})

const mapDispatchToProps = dispatch => bindActionCreators({
  attemptSignInWithGoogle, cancelGoogleAuth, fetchUsers
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(App)
