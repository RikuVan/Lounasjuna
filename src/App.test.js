import React from 'react'
import {App} from './App'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

test('Restaurant card snapshot test', () => {
  console.error = jest.genMockFn()
  const component = shallow(<App auth={{status: 'ANONYMOUS'}} />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
