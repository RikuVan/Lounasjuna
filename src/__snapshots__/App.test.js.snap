exports[`test Restaurant card snapshot test 1`] = `
<BrowserRouter>
  <div
    className="App">
    <div
      className="App-header">
      <Link
        replace={false}
        to="/">
        <h1
          className="Logo">
          <div
            className="Logo-train">
            ğŸš‚
          </div>
          <div
            className="Logo-text">
            Lounasjuna
          </div>
        </h1>
      </Link>
    </div>
    <div
      className="App-subheader">
      <div
        className="App-subheader-links">
        <Route
          exact={true}
          path="/"
          render={[Function]} />
        <Route
          path="/lounaspaikka/"
          render={[Function]} />
      </div>
      <CurrentUser
        status="ANONYMOUS" />
      <SignInOrOut
        loading={false}
        onClickHandler={[Function]}
        type="SignIn" />
    </div>
    <div
      className="App-content">
      <Connect(Notifier) />
      <Switch>
        <Route
          exact={true}
          path="/"
          render={[Function]} />
        <Route
          path="/lounaspaikka/uusi"
          render={[Function]} />
        <Route
          path="/"
          render={[Function]} />
      </Switch>
    </div>
  </div>
</BrowserRouter>
`;
